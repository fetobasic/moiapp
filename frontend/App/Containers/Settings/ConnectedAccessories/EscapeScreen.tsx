import React, { useState } from 'react';
import { ScrollView, StyleSheet, Text, View } from 'react-native';
import { ApplicationStyles, Colors, Fonts, Metrics } from 'App/Themes';
import {
  Button,
  CustomDropdown,
  HeaderSimple as Header,
  InformationRow as Row,
  SectionWithTitle,
  StyledRangeSlider,
} from 'App/Components';
import { NativeStackScreenProps } from '@react-navigation/native-stack';
import { SettingsStackParamList } from 'App/Types/NavigationStackParamList';
import { format } from 'date-fns';
import { useDispatch, useSelector } from 'react-redux';
import { getCurrentDevice } from 'App/Store/Devices/selectors';
import { navigationGoBack } from 'App/Navigation/AppNavigation';
import BrightnessLessIcon from 'App/Images/Icons/brightnessLess.svg';
import BrightnessMoreIcon from 'App/Images/Icons/brightnessMore.svg';
import { WithTopBorder } from 'App/Hoc';
import { devicesActions } from 'App/Store/Devices';
import { update } from 'App/Services/ConnectionControler';
import { Yeti6GState } from 'App/Types/Yeti';

type Props = NativeStackScreenProps<SettingsStackParamList, 'EscapeScreen'>;

const timeouts = [
  { value: 30, label: '30 Seconds' },
  { value: 60, label: '1 Minute' },
  { value: 300, label: '5 Minutes' },
  { value: 900, label: '15 Minutes' },
  { value: 1800, label: '30 Minutes' },
  { value: 3600, label: '1 Hour' },
  { value: 0, label: 'Never' },
];

const EscapeScreen = ({ navigation, route }: Props) => {
  const { id } = route.params.escapeScreen;
  const device = useSelector(getCurrentDevice(route.params.device.thingName || '')) as Yeti6GState;
  const [brightness, setBrightness] = useState(device?.config?.xNodes[id]?.dsp?.brt || 0);
  const [isChanged, setChanged] = useState<boolean>(false);
  const dispatch = useDispatch();
  const [timeout, setTimeout] = useState<(typeof timeouts)[number]>(
    timeouts.find((item) => item.value === device?.config?.xNodes[id]?.dsp?.tOut) || { value: 0, label: 'Never' },
  );

  const onSaveHandler = () => {
    update({
      thingName: device.thingName || '',
      method: 'config',
      // @ts-ignore TODO: we need to use complex and recursive Partial generic of stateObject type
      stateObject: { state: { desired: { xNodes: { [id]: { dsp: { tOut: timeout?.value, brt: brightness } } } } } },
      isDirectConnection: Boolean(device?.isDirectConnection),
      updateDeviceState: (_thingName, data) =>
        dispatch(devicesActions.devicesAddUpdate.request({ thingName: _thingName, data })),
    });

    //update local state via dispatch action, temporary solution to reflect changes on the screen sooner
    dispatch(
      devicesActions.devicesAddUpdate.request({
        thingName: device.thingName || '',
        data: {
          config: {
            // @ts-ignore TODO: we need to use complex and recursive Partial generic of Yeti6GState type
            xNodes: {
              [id]: {
                dsp: {
                  tOut: timeout?.value,
                  brt: brightness,
                },
              },
            },
          },
        },
      }),
    );
    setChanged(false);
    navigationGoBack();
  };

  const onLimitChangeHandler = (backup: number, _: any, fromUser: boolean) => {
    if (!fromUser) {
      return;
    }
    setBrightness(backup);
  };

  return (
    <View style={ApplicationStyles.mainContainer}>
      <Header
        title="Escape Screen"
        isChangeSaved={isChanged}
        cbSave={onSaveHandler}
        subTitle={
          device?.device?.xNodes[id]?.tConn ? format(device?.device?.xNodes[id]?.tConn * 1000, "h:mm bbb 'on' PP") : ''
        }
      />
      <ScrollView style={styles.container}>
        <SectionWithTitle>
          <Row title="Model" description="Escape Screen" descriptionStyle={{ color: Colors.gray }} />
          <Row
            title="Firmware Version"
            subTitle="Updated by Yeti."
            description={`Version ${device?.device?.xNodes[id]?.fw}`}
            descriptionStyle={{ color: Colors.gray }}
          />
        </SectionWithTitle>
        <Row
          contentStyle={styles.inputLimits}
          title="Power+ Input Limit"
          description={`${device?.config?.ports?.acIn?.plus?.aLmt} amps`}
          onPress={() => navigation.navigate('PowerInputLimit', { device })}
        />
        <WithTopBorder>
          <Text style={Fonts.font.base.bodyTwo}>
            <Text style={ApplicationStyles.textGreen}>{brightness}%</Text> - Brightness
          </Text>
          <View style={styles.sliderWrapper}>
            <BrightnessLessIcon />
            <StyledRangeSlider disableRange onValueChanged={onLimitChangeHandler} low={brightness} />
            <BrightnessMoreIcon />
          </View>
        </WithTopBorder>
        <CustomDropdown
          containerStyle={styles.dropDown}
          placeholder="Screen Timeout"
          data={timeouts}
          header="Screen Timeout"
          value={timeout}
          onChange={(item) => {
            setTimeout(item as (typeof timeouts)[number]);
            setChanged(true);
          }}
        />
      </ScrollView>
      <Button style={styles.saveBtn} title="SAVE" onPress={onSaveHandler} />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    paddingHorizontal: Metrics.marginSection,
    paddingVertical: Metrics.halfMargin,
  },
  sliderWrapper: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  dropDown: {
    marginTop: Metrics.halfMargin,
  },
  inputLimits: {
    marginBottom: 4,
  },
  saveBtn: {
    marginHorizontal: Metrics.baseMargin,
    marginVertical: Metrics.marginSection,
  },
});

export default EscapeScreen;
