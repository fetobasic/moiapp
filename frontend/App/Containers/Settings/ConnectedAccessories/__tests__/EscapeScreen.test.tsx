import React from 'react';
import { render } from '@testing-library/react-native';
import { NavigationContainer } from '@react-navigation/native';
import { Provider } from 'react-redux';
import createMockStore from 'redux-mock-store';
import { mockedState } from 'App/Fixtures/mocks/mockedState';
import EscapeScreen from '../EscapeScreen';
import { yeti6GState } from 'App/Fixtures/yeti6GState';

const mockStore = createMockStore([]);
const store = mockStore({
  ...mockedState,
});

describe('EscapeScreen', () => {
  test('renders screen', () => {
    const props = {
      navigation: {
        navigate: jest.fn() as any,
      },
      route: { params: { escapeScreen: { id: '123' }, device: yeti6GState } },
    } as any;

    const tree = render(
      <Provider store={store}>
        <NavigationContainer>
          <EscapeScreen {...props} />
        </NavigationContainer>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
