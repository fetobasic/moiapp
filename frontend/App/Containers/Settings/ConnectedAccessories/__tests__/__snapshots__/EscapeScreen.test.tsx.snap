// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EscapeScreen renders screen 1`] = `
<View
  style={
    {
      "backgroundColor": "rgba(27, 27, 27, 1)",
      "flex": 1,
    }
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "rgba(27, 27, 27, 1)",
          "borderBottomLeftRadius": 24,
          "borderBottomRightRadius": 24,
          "elevation": 12,
          "shadowColor": "rgba(0, 0, 0, 1)",
          "shadowOffset": {
            "height": 4,
            "width": 4,
          },
          "shadowOpacity": 0.2,
          "shadowRadius": 12,
          "zIndex": 1,
        },
        {
          "paddingTop": 0,
        },
      ]
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 60,
          "justifyContent": "center",
          "marginHorizontal": 24,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        hitSlop={20}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "opacity": 1,
            },
            {
              "left": 0,
              "padding": 10,
              "position": "absolute",
            },
          ]
        }
      >
        <SvgMock />
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "marginHorizontal": 48,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            {
              "color": "rgba(255, 255, 255, 1)",
              "fontFamily": "RobotoCondensed-Medium",
              "fontSize": 20,
              "fontWeight": "500",
              "lineHeight": 32,
            }
          }
        >
          Escape Screen
        </Text>
      </View>
    </View>
  </View>
  <RCTScrollView
    style={
      {
        "paddingHorizontal": 16,
        "paddingVertical": 12,
      }
    }
  >
    <View>
      <View
        style={
          [
            {
              "borderColor": "rgba(54, 54, 54, 1)",
              "borderRadius": 8,
              "borderWidth": 1,
              "marginVertical": 12,
              "paddingBottom": 6,
              "paddingHorizontal": 8,
              "paddingTop": 12,
              "position": "relative",
            },
            undefined,
            undefined,
          ]
        }
      >
        <View
          style={
            [
              {
                "paddingHorizontal": 12,
                "paddingVertical": 15,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              [
                {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                },
                undefined,
              ]
            }
          >
            <View
              style={
                [
                  {
                    "flex": 1,
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
            >
              <Text
                ellipsizeMode="tail"
                numberOfLines={1}
                style={
                  [
                    {
                      "color": "rgba(255, 255, 255, 0.87)",
                      "fontFamily": "Roboto-Medium",
                      "fontSize": 16,
                      "fontWeight": "400",
                      "lineHeight": 24,
                      "maxWidth": 210,
                    },
                    undefined,
                  ]
                }
              >
                Model
              </Text>
            </View>
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  false,
                ]
              }
            >
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "opacity": 1,
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  ellipsizeMode="clip"
                  numberOfLines={1}
                  style={
                    [
                      {
                        "color": "rgba(255, 255, 255, 0.87)",
                        "fontFamily": "Roboto-Medium",
                        "fontSize": 16,
                        "fontWeight": "400",
                        "lineHeight": 24,
                        "maxWidth": 210,
                      },
                      {
                        "color": "rgba(147, 147, 147, 1)",
                      },
                    ]
                  }
                >
                  Escape Screen
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            [
              {
                "paddingHorizontal": 12,
                "paddingVertical": 15,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              [
                {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                },
                undefined,
              ]
            }
          >
            <View
              style={
                [
                  {
                    "flex": 1,
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
            >
              <Text
                ellipsizeMode="tail"
                numberOfLines={1}
                style={
                  [
                    {
                      "color": "rgba(255, 255, 255, 0.87)",
                      "fontFamily": "Roboto-Medium",
                      "fontSize": 16,
                      "fontWeight": "400",
                      "lineHeight": 24,
                      "maxWidth": 210,
                    },
                    undefined,
                  ]
                }
              >
                Firmware Version
              </Text>
              <Text
                ellipsizeMode="tail"
                numberOfLines={1}
                style={
                  [
                    {
                      "color": "rgba(147, 147, 147, 1)",
                      "fontFamily": "Roboto-Medium",
                      "fontSize": 14,
                      "fontWeight": "400",
                      "lineHeight": 20,
                    },
                    undefined,
                    undefined,
                  ]
                }
              >
                Updated by Yeti.
              </Text>
            </View>
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  {
                    "alignItems": "flex-start",
                  },
                ]
              }
            >
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "opacity": 1,
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  ellipsizeMode="clip"
                  numberOfLines={1}
                  style={
                    [
                      {
                        "color": "rgba(255, 255, 255, 0.87)",
                        "fontFamily": "Roboto-Medium",
                        "fontSize": 16,
                        "fontWeight": "400",
                        "lineHeight": 24,
                        "maxWidth": 210,
                      },
                      {
                        "color": "rgba(147, 147, 147, 1)",
                      },
                    ]
                  }
                >
                  Version undefined
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          [
            {
              "borderRadius": 16,
              "elevation": 2,
              "marginBottom": 8,
              "shadowColor": "rgba(0, 0, 0, 0.4)",
              "shadowOffset": {
                "height": 4,
                "width": 0,
              },
              "shadowOpacity": 0.7,
              "shadowRadius": 7,
            },
            undefined,
            undefined,
          ]
        }
      >
        <View
          style={
            [
              {
                "borderRadius": 16,
                "marginHorizontal": 1,
                "overflow": "hidden",
                "paddingTop": 1,
                "position": "relative",
              },
              undefined,
              undefined,
            ]
          }
        >
          <RNSVGSvgView
            bbHeight="100%"
            bbWidth="100%"
            focusable={false}
            height="100%"
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                {
                  "flex": 0,
                  "height": "100%",
                  "width": "100%",
                },
              ]
            }
            width="100%"
          >
            <RNSVGGroup
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
            >
              <RNSVGDefs>
                <RNSVGLinearGradient
                  gradient={
                    [
                      0,
                      -13224394,
                      1,
                      1776411,
                    ]
                  }
                  gradientTransform={null}
                  gradientUnits={0}
                  name="grad"
                  x1="0"
                  x2="0"
                  y1="0"
                  y2="0.5"
                />
              </RNSVGDefs>
              <RNSVGRect
                fill={
                  {
                    "brushRef": "grad",
                    "type": 1,
                  }
                }
                height="100%"
                propList={
                  [
                    "fill",
                  ]
                }
                width="100%"
                x="0"
                y="0"
              />
            </RNSVGGroup>
          </RNSVGSvgView>
          <View
            style={
              [
                {
                  "backgroundColor": "rgba(27, 27, 27, 1)",
                  "borderRadius": 16,
                  "paddingHorizontal": 12,
                  "paddingVertical": 15,
                },
                {
                  "marginBottom": 4,
                },
                undefined,
                undefined,
              ]
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                [
                  {
                    "opacity": 1,
                  },
                  [
                    {
                      "zIndex": 1000,
                    },
                    undefined,
                    undefined,
                  ],
                ]
              }
              testID="withPressable"
            >
              <View
                style={
                  [
                    {
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                    },
                    undefined,
                  ]
                }
              >
                <View
                  style={
                    [
                      {
                        "flex": 1,
                        "justifyContent": "center",
                      },
                      undefined,
                    ]
                  }
                >
                  <Text
                    ellipsizeMode="tail"
                    numberOfLines={1}
                    style={
                      [
                        {
                          "color": "rgba(255, 255, 255, 0.87)",
                          "fontFamily": "Roboto-Medium",
                          "fontSize": 16,
                          "fontWeight": "400",
                          "lineHeight": 24,
                          "maxWidth": 210,
                        },
                        undefined,
                      ]
                    }
                  >
                    Power+ Input Limit
                  </Text>
                </View>
                <View
                  style={
                    [
                      {
                        "alignItems": "center",
                        "flexDirection": "row",
                      },
                      false,
                    ]
                  }
                >
                  <View
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      [
                        {
                          "opacity": 1,
                        },
                        undefined,
                      ]
                    }
                  >
                    <Text
                      ellipsizeMode="clip"
                      numberOfLines={1}
                      style={
                        [
                          {
                            "color": "rgba(255, 255, 255, 0.87)",
                            "fontFamily": "Roboto-Medium",
                            "fontSize": 16,
                            "fontWeight": "400",
                            "lineHeight": 24,
                            "maxWidth": 210,
                          },
                          undefined,
                        ]
                      }
                    >
                      undefined amps
                    </Text>
                  </View>
                  <SvgMock
                    style={
                      [
                        {
                          "marginLeft": 16,
                        },
                        undefined,
                      ]
                    }
                  />
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          [
            {
              "borderRadius": 16,
              "elevation": 2,
              "marginBottom": 8,
              "shadowColor": "rgba(0, 0, 0, 0.4)",
              "shadowOffset": {
                "height": 4,
                "width": 0,
              },
              "shadowOpacity": 0.7,
              "shadowRadius": 7,
            },
            undefined,
            undefined,
          ]
        }
      >
        <View
          style={
            [
              {
                "borderRadius": 16,
                "marginHorizontal": 1,
                "overflow": "hidden",
                "paddingTop": 1,
                "position": "relative",
              },
              undefined,
              undefined,
            ]
          }
        >
          <RNSVGSvgView
            bbHeight="100%"
            bbWidth="100%"
            focusable={false}
            height="100%"
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                {
                  "flex": 0,
                  "height": "100%",
                  "width": "100%",
                },
              ]
            }
            width="100%"
          >
            <RNSVGGroup
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
            >
              <RNSVGDefs>
                <RNSVGLinearGradient
                  gradient={
                    [
                      0,
                      -13224394,
                      1,
                      1776411,
                    ]
                  }
                  gradientTransform={null}
                  gradientUnits={0}
                  name="grad"
                  x1="0"
                  x2="0"
                  y1="0"
                  y2="0.5"
                />
              </RNSVGDefs>
              <RNSVGRect
                fill={
                  {
                    "brushRef": "grad",
                    "type": 1,
                  }
                }
                height="100%"
                propList={
                  [
                    "fill",
                  ]
                }
                width="100%"
                x="0"
                y="0"
              />
            </RNSVGGroup>
          </RNSVGSvgView>
          <View
            style={
              [
                {
                  "backgroundColor": "rgba(27, 27, 27, 1)",
                  "borderRadius": 16,
                  "paddingHorizontal": 12,
                  "paddingVertical": 15,
                },
                undefined,
                undefined,
                undefined,
              ]
            }
          >
            <Text
              style={
                {
                  "color": "rgba(255, 255, 255, 0.87)",
                  "fontFamily": "Roboto-Medium",
                  "fontSize": 16,
                  "fontWeight": "400",
                  "lineHeight": 24,
                }
              }
            >
              <Text
                style={
                  {
                    "color": "rgba(191, 210, 43, 1)",
                  }
                }
              >
                0
                %
              </Text>
               - Brightness
            </Text>
            <View
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <SvgMock />
              <View
                style={
                  [
                    {
                      "flex": 1,
                      "flexDirection": "row",
                      "position": "relative",
                    },
                    undefined,
                  ]
                }
              >
                <View
                  style={
                    {
                      "flex": 1,
                    }
                  }
                >
                  <View
                    onLayout={[Function]}
                    testID="styledRangeSlider"
                  >
                    <View
                      onLayout={[Function]}
                      style={
                        [
                          {
                            "alignItems": "flex-start",
                          },
                          {
                            "top": 0,
                          },
                        ]
                      }
                    />
                    <View
                      onLayout={[Function]}
                      style={
                        {
                          "alignItems": "center",
                          "flexDirection": "row",
                          "justifyContent": "flex-start",
                        }
                      }
                    >
                      <View
                        style={
                          [
                            {
                              "alignItems": "center",
                              "bottom": 0,
                              "flexDirection": "row",
                              "left": 0,
                              "position": "absolute",
                              "right": 0,
                              "top": 0,
                            },
                            {
                              "marginHorizontal": 0,
                            },
                          ]
                        }
                      >
                        <View
                          style={
                            [
                              {
                                "backgroundColor": "rgba(191, 210, 43, 1)",
                                "borderRadius": 18,
                                "height": 3,
                                "overflow": "hidden",
                              },
                              {
                                "backgroundColor": "rgba(54, 54, 54, 1)",
                              },
                              {
                                "width": "100%",
                              },
                            ]
                          }
                        />
                        <View
                          collapsable={false}
                          style={
                            {
                              "left": 0,
                              "position": "absolute",
                              "right": 0,
                            }
                          }
                        >
                          <View
                            style={
                              [
                                {
                                  "backgroundColor": "rgba(191, 210, 43, 1)",
                                  "borderRadius": 18,
                                  "height": 3,
                                  "overflow": "hidden",
                                },
                                {
                                  "backgroundColor": "rgba(191, 210, 43, 1)",
                                },
                                {
                                  "width": "100%",
                                },
                              ]
                            }
                          />
                        </View>
                      </View>
                      <View
                        collapsable={false}
                        onLayout={[Function]}
                        style={
                          {
                            "transform": [
                              {
                                "translateX": 0,
                              },
                            ],
                          }
                        }
                      >
                        <View
                          style={
                            [
                              {
                                "borderRadius": 14,
                                "borderWidth": 1,
                                "height": 24,
                                "width": 24,
                              },
                              {
                                "backgroundColor": "rgba(191, 210, 43, 1)",
                                "borderColor": "rgba(191, 210, 43, 1)",
                              },
                            ]
                          }
                        />
                      </View>
                      <View
                        collapsable={false}
                        onMoveShouldSetResponder={[Function]}
                        onMoveShouldSetResponderCapture={[Function]}
                        onResponderEnd={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderReject={[Function]}
                        onResponderRelease={[Function]}
                        onResponderStart={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        onStartShouldSetResponderCapture={[Function]}
                        style={
                          {
                            "bottom": 0,
                            "left": 0,
                            "position": "absolute",
                            "right": 0,
                            "top": 0,
                          }
                        }
                      />
                    </View>
                  </View>
                </View>
              </View>
              <SvgMock />
            </View>
          </View>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "borderColor": "rgba(54, 54, 54, 1)",
              "borderRadius": 8,
              "borderWidth": 1,
              "paddingHorizontal": 12,
              "paddingVertical": 16,
              "position": "relative",
            },
            {
              "marginTop": 12,
            },
          ]
        }
        testID="customDropdownPress"
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              {
                "color": "rgba(255, 255, 255, 0.87)",
                "fontFamily": "Roboto-Medium",
                "fontSize": 14,
                "fontWeight": "400",
                "lineHeight": 20,
              }
            }
          >
            Never
          </Text>
          <SvgMock />
        </View>
        <Text
          style={
            {
              "backgroundColor": "rgba(27, 27, 27, 1)",
              "color": "rgba(191, 210, 43, 1)",
              "fontFamily": "Roboto-Medium",
              "fontSize": 12,
              "fontWeight": "300",
              "left": 7,
              "lineHeight": 16,
              "paddingHorizontal": 7,
              "position": "absolute",
              "top": -9,
            }
          }
        >
          Screen Timeout
        </Text>
      </View>
    </View>
  </RCTScrollView>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "opacity": 1,
        },
        [
          {
            "elevation": 12,
            "shadowColor": "rgba(0, 0, 0, 1)",
            "shadowOffset": {
              "height": 4,
              "width": 4,
            },
            "shadowOpacity": 0.2,
            "shadowRadius": 12,
          },
          {
            "marginHorizontal": 24,
            "marginVertical": 16,
          },
        ],
      ]
    }
  >
    <View
      style={
        {
          "borderRadius": 24,
          "overflow": "hidden",
        }
      }
    >
      <RNSVGSvgView
        bbHeight="48"
        bbWidth="100%"
        focusable={false}
        height={48}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 48,
              "width": "100%",
            },
          ]
        }
        width="100%"
      >
        <RNSVGGroup
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
        >
          <RNSVGDefs>
            <RNSVGLinearGradient
              gradient={
                [
                  0,
                  -13224394,
                  1,
                  1776411,
                ]
              }
              gradientTransform={null}
              gradientUnits={0}
              name="grad"
              x1="0"
              x2="0"
              y1="0"
              y2="0.5"
            />
          </RNSVGDefs>
          <RNSVGRect
            fill={
              {
                "brushRef": "grad",
                "type": 1,
              }
            }
            height="48"
            propList={
              [
                "fill",
              ]
            }
            width="100%"
            x="0"
            y="0"
          />
          <View
            style={
              [
                {
                  "borderRadius": 24,
                  "marginHorizontal": 1,
                  "marginTop": 1,
                },
                {
                  "backgroundColor": "rgba(191, 210, 43, 1)",
                  "height": 46,
                },
              ]
            }
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "paddingVertical": 12,
                  },
                  undefined,
                ]
              }
            >
              <View
                style={
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                  }
                }
              >
                <Text
                  style={
                    [
                      {
                        "color": "rgba(255, 255, 255, 1)",
                        "fontFamily": "Roboto-Medium",
                        "fontSize": 14,
                        "fontWeight": "700",
                        "lineHeight": 20,
                        "textTransform": "uppercase",
                      },
                      {
                        "color": "rgba(27, 27, 27, 1)",
                      },
                      undefined,
                    ]
                  }
                >
                  SAVE
                </Text>
              </View>
            </View>
          </View>
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
  </View>
</View>
`;
